File: 0014-longest-common-prefix/0014-longest-common-prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        s = set(strs)
        hashMap = {}
        menor = strs[0]
        string = ""
        # Acha a menor palavra dentro do conjunto
        for palavra in strs:
            if len(palavra) < len(menor):
                menor = palavra

        for index in range(len(menor)):
            prefixo = menor[:index+1]
            if not all(palavra.startswith(prefixo) for palavra in s):
                return menor[:index]
        
        return menor

File: 0014-longest-common-prefix/0014-longest-common-prefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        if (strs == null || strs.length == 0) return "";

        String prefix = "";
        String firstWord = strs[0];

        for (int i = 0; i < firstWord.length(); i++) {
            char currentChar = firstWord.charAt(i);

            for (int j = 1; j < strs.length; j++) {
                if (i >= strs[j].length() || strs[j].charAt(i) != currentChar) {
                    return prefix;
                }
            }
            prefix += currentChar;
        }
        return prefix;
    }
}